package com.example.recipeapp.ui.components.layout

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Close
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import com.example.recipeapp.models.database.Food
import com.example.recipeapp.models.database.Log
import com.example.recipeapp.viewmodels.LogsScreenViewModel

/**
 * A composable function that displays a row representing a log.
 */
@Composable
fun LogRow(log: Log, viewModel: LogsScreenViewModel) {
    var food: Food? by remember { mutableStateOf(null) }

    LaunchedEffect(log) {
        food = viewModel.fetchFoodById(log.foodId)
    }

    Row(
        modifier = Modifier.fillMaxWidth(),
        horizontalArrangement = Arrangement.SpaceBetween,
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Column {
            Text(food?.name ?: "Loading...", style = MaterialTheme.typography.titleLarge,)
            NutrientRow(food = food)
        }
        Row {
            Text(text = "${log.amount}")
            IconButton(onClick = { log.id?.let { viewModel.deleteLog(it) } }) {
                Icon(imageVector = Icons.Filled.Close, contentDescription = "Close")
            }
        }
    }
}
